src/Spreadsheet.ts

```diff
@@ -2,34 +2,72 @@
 ///
 
 import * as _Promise from "bluebird";
+import * as fs from "fs-extra";
 import { auth, JWT } from "google-auth-library";
-import { google } from "googleapis";
+import { google, sheets_v4 } from "googleapis";
 import * as _ from "lodash";
+import * as xlsx from "node-xlsx";
+import * as path from "path";
 import { IVoxaSheet, SheetTypes } from "./VoxaSheet";
 
-export async function transform(options: any, authKeys: {}): Promise<IVoxaSheet[]> {
-  const client = auth.fromJSON(authKeys) as JWT;
-  client.scopes = ["https://www.googleapis.com/auth/spreadsheets.readonly"];
-  const sheets = google.sheets("v4");
-  const readSpreadsheet: any = _Promise.promisify(sheets.spreadsheets.get, { context: sheets });
-  const readSheetTab: any = _Promise.promisify(sheets.spreadsheets.values.get, { context: sheets });
-  const spreadsheetsId = options.spreadsheets as string[];
+const sheets = google.sheets("v4");
+const readSpreadsheet: any = _Promise.promisify(sheets.spreadsheets.get, { context: sheets });
+const readSheetTab: any = _Promise.promisify(sheets.spreadsheets.values.get, { context: sheets });
 
-  // TODO: error on getting spreadsheet
-  let spreadsheetResp = [] as any[];
+function findLocalFiles(spreadsheet: string): string[] {
+  const fsStats = fs.lstatSync(spreadsheet);
+  if (fsStats.isDirectory()) {
+    return fs
+      .readdirSync(spreadsheet)
+      .filter(f => _.includes(f, "xlsx"))
+      .map(f => path.join(spreadsheet, f));
+  }
+  return [spreadsheet];
+}
 
-  try {
-    spreadsheetResp = await _Promise.all(
-      spreadsheetsId.map((spreadsheetId: string) =>
-        readSpreadsheet({ auth: client, spreadsheetId })
-      )
-    );
-  } catch (e) {
-    throw Error("User doesn't have access to the spreadsheet");
+function readFileCreateWorkbook(f: string) {
+  const workbook = xlsx.parse(f);
+  const spreadsheetTitle = _.last(f.split("/"));
+  const spreadsheetId = f;
+  if (!_.isString(spreadsheetTitle)) {
+    return undefined;
   }
+  return workbook.map(book => ({
+    spreadsheetId,
+    spreadsheetTitle,
+    sheetTitle: book.name,
+    type: "none",
+    data: book.data
+  }));
+}
 
-  spreadsheetResp = _.chain(spreadsheetResp)
-    .map((spreadsheet: string, index: number) => {
+function refactorExcelData(sheet: IVoxaSheet) {
+  sheet.data = _.chain(sheet)
+    .get("data")
+    .map((next, index: number, arr) => {
+      if (index === 0) {
+        return next;
+      }
+      const head = arr[0];
+
+      const extraColumns = head.length - next.length;
+      if (extraColumns > 0) {
+        next = _.concat(next, _.fill(Array(extraColumns), undefined));
+      }
+      return next;
+    })
+    .reduce(rowFormatted, [] as any[])
+    .drop()
+    .value();
+  return sheet;
+}
+
+function refactorSpreadsheet(
+  spreadsheetsId: string[],
+  spreadsheetResp: sheets_v4.Resource$Spreadsheets$Sheets[] | any[]
```

Comment:
why the `any`? also, what does `refactorSpreadsheet` mean? can we have a return type?